crates/p2p/src/behaviour.rs

```diff
@@ -182,6 +182,17 @@ impl NetworkBehaviour for Behaviour {
             return Err(ConnectionDenied::new(anyhow!("peer without IP")));
         };
 
+        // If the peer is not in the IP whitelist, disconnect.
+        if !self
+            .limits
+            .ip_whitelist
+            .iter()
+            .any(|net| net.contains(&peer_ip))
+        {
+            tracing::debug!(%connection_id, "Disconnected peer not in IP whitelist");
```

Comment:
nit: connection_id not so informative, maybe add peer_ip here since it's available anyway?